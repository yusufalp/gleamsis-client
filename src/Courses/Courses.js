import React from 'react';
import { Link } from 'react-router-dom';
import DataContext from '../DataContext';

class Courses extends React.Component {
  static contextType = DataContext;
  handleAddStudent() {
    console.log("Add Student")
  }
  render() {
    let course = this.context.courses.find(course => course.id === Number(this.props.match.params.id))
    let teacher = this.context.teachers.find(teacher => teacher.id === course.teacher_id)
    let students = this.context.students.filter(students => students.course_id === course.id)
    return (
      <div>
        <h3>Hello {teacher.firstName}, this is {course.name} class</h3>
        <table>
          <thead>
            <tr>
              <th>First Name</th>
              <th>Last Name</th>
              <th>Grade</th>
            </tr>
          </thead>
          {students.map((student, i) =>
            <tbody key={i}>
              <tr>
                <td><Link to={`/students/${student.id}`}>{student.firstName}</Link><hr /></td>
                <td><Link to={`/students/${student.id}`}>{student.lastName}</Link><hr /></td>
                <td>{student.grade}<hr /></td>
              </tr>
            </tbody>
          )}
        </table>
        <p className={`${students.length === 0 ? "" : "hidden"}`}>No students are registered for this class</p>
        <Link to={'/AddStudent'} className="btn-teacher">Add Student</Link>
      </div>
    )
  }
}

export default Courses;