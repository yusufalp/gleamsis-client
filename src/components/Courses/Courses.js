import React from 'react';
import { Link } from 'react-router-dom';
import DataContext from '../DataContext';

class Courses extends React.Component {
  static contextType = DataContext;
  render() {
    let course = [];
    let teacher = [];
    let students = [];
    course = this.context.courses.find(course => course.id === Number(this.props.match.params.id)) || {};
    teacher = this.context.teachers.find(teacher => teacher.id === course.teacher_id) || {};
    students = this.context.students.filter(students => students.course_id === course.id);
    return (
      <div className='dash'>
        <h3>Hello {teacher.first_name}, this is {course.name} class</h3>
        <table className='card-display'>
          <thead>
            <tr>
              <th>First Name</th>
              <th>Last Name</th>
              <th>Grade</th>
            </tr>
          </thead>
          {students.map((student, i) =>
            <tbody key={i}>
              <tr>
                <td><Link to={`/students/${student.id}`}>{student.first_name}</Link><hr /></td>
                <td><Link to={`/students/${student.id}`}>{student.last_name}</Link><hr /></td>
                <td>{student.grade}<hr /></td>
              </tr>
            </tbody>
          )}
        </table>
        <p className={`${students.length === 0 ? '' : 'hidden'}`}>No students are registered for this class</p>
        <Link to={'/add-student'} className='btn-add'>Add Student</Link>
      </div>
    );
  }
}

export default Courses;