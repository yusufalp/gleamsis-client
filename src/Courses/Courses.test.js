import React from 'react';
import ReactDOM from 'react-dom';
import Courses from './Courses';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Courses component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Courses />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Courses />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})