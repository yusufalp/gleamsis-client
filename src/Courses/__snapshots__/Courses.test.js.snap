// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses component renders the UI as expected 1`] = `
<div
  className="dash"
>
  <h3>
    Hello 
    , this is 
     class
  </h3>
  <table>
    <thead>
      <tr>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Grade
        </th>
      </tr>
    </thead>
  </table>
  <p
    className=""
  >
    No students are registered for this class
  </p>
  <a
    className="btn-add"
    href="/add-student"
    onClick={[Function]}
  >
    Add Student
  </a>
</div>
`;
